crates/fluvio-benchmark/src/benchmark_config/mod.rs

```diff
@@ -0,0 +1,257 @@
+            && self.num_records_per_producer_worker_per_batch
+                == other.num_records_per_producer_worker_per_batch
+            && self.producer_batch_size == other.producer_batch_size
+            && self.producer_queue_size == other.producer_queue_size
+            && self.producer_linger == other.producer_linger
+            && self.producer_server_timeout == other.producer_server_timeout
+            && self.producer_compression == other.producer_compression
+            && self.consumer_max_bytes == other.consumer_max_bytes
+            && self.num_concurrent_producer_workers == other.num_concurrent_producer_workers
+            && self.num_concurrent_consumers_per_partition
+                == other.num_concurrent_consumers_per_partition
+            && self.num_partitions == other.num_partitions
+            && self.record_size == other.record_size
+            && self.record_key_allocation_strategy == other.record_key_allocation_strategy
+            && self.matrix_name == other.matrix_name
+    }
+}
+
+impl Hash for BenchmarkConfig {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        // We don't hash the topic name, current_profile, or timestamp
+        self.worker_timeout.hash(state);
+        self.num_samples.hash(state);
+        self.duration_between_samples.hash(state);
+        self.num_records_per_producer_worker_per_batch.hash(state);
+        self.producer_batch_size.hash(state);
+        self.producer_queue_size.hash(state);
+        self.producer_linger.hash(state);
+        self.producer_server_timeout.hash(state);
+        self.producer_compression.hash(state);
+        self.consumer_max_bytes.hash(state);
+        self.num_concurrent_producer_workers.hash(state);
+        self.num_concurrent_consumers_per_partition.hash(state);
+        self.num_partitions.hash(state);
+        self.record_size.hash(state);
+        self.record_key_allocation_strategy.hash(state);
+        self.matrix_name.hash(state);
+    }
+}
+
+impl BenchmarkConfig {
+    pub fn total_number_of_messages_produced_per_batch(&self) -> u64 {
+        self.num_records_per_producer_worker_per_batch as u64
+            * self.num_concurrent_producer_workers as u64
+    }
+
+    pub fn number_of_expected_times_each_message_consumed(&self) -> u64 {
+        self.num_concurrent_consumers_per_partition
+    }
+
+    pub fn to_markdown(&self) -> String {
+        let mut md = String::new();
+        md.push_str("**Config**\n");
+        md.push_str("```yaml");
+        md.push_str(&serde_yaml::to_string(self).unwrap());
+        md.push_str("```");
+        md
+    }
+}
+
+pub fn generate_new_topic_name() -> String {
+    let mut rng = thread_rng();
+    let chars: String = (0..15)
+        .map(|_| rng.sample(Uniform::new(b'a', b'z')) as char)
+        .collect();
+    format!("benchmarking-{}", chars)
+}
+
+#[derive(Clone)]
+pub struct BenchmarkBuilder {
+    pub shared_config: SharedConfig,
```

Comment:
There is quite overlap with config. Is there reason for this?  You might want to look at `derive_builder` if you want to create builder

Reply:
yes switching to `derive_builder` is much better.